modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java

```diff
@@ -0,0 +1,305 @@
+    }
+
+
+    /**
+     * Delete the policy.
+     *
+     * @param id the id of policy.
+     */
+    public void deletePolicy(UUID id) {
+        PolicyEntity existingEntity = policyStorage.findPolicyById(id);
+        if (Objects.isNull(existingEntity)) {
+            String errorMsg = String.format("The policy with id %s not found.", id);
+            throw new PolicyNotFoundException(errorMsg);
+        }
+        Optional<String> userIdOptional = identityProviderManager.getCurrentLoginUserId();
+        if (!StringUtils.equals(userIdOptional.orElse(null), existingEntity.getUserId())) {
+            throw new AccessDeniedException(
+                    "No permissions to delete policy belonging to other users.");
+        }
+        policyStorage.deletePolicyById(id);
+    }
+
+    private void checkPolicyWhetherCreated(Csp csp, String policy) {
+
+        PolicyQueryRequest queryModel = new PolicyQueryRequest();
+        Optional<String> userIdOptional = identityProviderManager.getCurrentLoginUserId();
+        queryModel.setUserId(userIdOptional.orElse(null));
+        queryModel.setCsp(csp);
+        queryModel.setPolicy(policy);
+        List<PolicyEntity> policyEntityList = policyStorage.listPolicies(queryModel);
+        if (!CollectionUtils.isEmpty(policyEntityList)) {
+            String policyKey = policyEntityList.get(0).getId().toString();
+            String errMsg = String.format("The same policy already created by you for the Csp: %s."
+                    + " id: %s", csp, policyKey);
+            throw new PolicyDuplicateException(errMsg);
+        }
+
+    }
+
+
+    private PolicyVo conventToPolicyVo(PolicyEntity policyEntity) {
+        if (Objects.nonNull(policyEntity)) {
+            PolicyVo policyVo = new PolicyVo();
+            BeanUtils.copyProperties(policyEntity, policyVo);
+            return policyVo;
+        }
+        return null;
+    }
+
+    private PolicyEntity conventToPolicyEntity(PolicyCreateRequest policyCreateRequest) {
+        PolicyEntity policyEntity = new PolicyEntity();
+        BeanUtils.copyProperties(policyCreateRequest, policyEntity);
+        Optional<String> userIdOptional = identityProviderManager.getCurrentLoginUserId();
+        policyEntity.setUserId(userIdOptional.orElse(null));
+        if (Objects.isNull(policyEntity.getEnabled())) {
+            policyEntity.setEnabled(true);
+        }
+        return policyEntity;
+    }
+
+
+    private void validatePolicy(String policy) {
+        boolean valid = true;
+        String errorMsg = "";
+        ValidatePolicyList validatePolicyList = new ValidatePolicyList();
+        validatePolicyList.setPolicyList(List.of(policy));
+        try {
+            ValidateResponse validateResponse =
+                    policiesValidateApi.validatePoliciesPost(validatePolicyList);
+            log.info("Validate policies response:{}", validateResponse.toString());
+            if (Objects.isNull(validateResponse.getIsSuccessful())
```

Comment:
validateResponse.getIsSuccessful( ) -> is never null since this is a mandatory field. If policy-man is returning null here sometimes, then it is better we fix it in policy-man.